Woo Onboarding: Add revenue dropdown

obenland opened the pull request:
#### Changes proposed in this Pull Request

* Adds revenue dropdown to business-info step.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Navigate to `http://calypso.localhost:3000/start/woocommerce-install/store-address?site=<siteId>`
* Fill up the fields specially the country/state one, select US - <Any State> and click continue
* On `Currently selling elsewhere? (optional)` dropdown select "Yes on another platform"
* Revenue Dropdown should appear and contain the same options as the woocommerce one.
* To try out different currencies, go back to the address step and select a different country.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #59837

obenland committed:
Woo Installer: Add business-info step

obenland committed:
Update with more dropdowns and improved navigation

obenland committed:
Update data handling

obenland committed:
Remove currency-related code

obenland committed:
Woo Onboarding: Add revenue dropdown

a8ci18n commented on client/signup/steps/woocommerce-install/step-business-info/revenue-options.js:
```diff
@@ -0,0 +1,124 @@
+	},
+	{
+		value: 'rather-not-say',
+		label: __( "I'd rather not say", 'woocommerce-admin' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'I'd rather not answer' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-revenue-dropdown)

lsl replied:
```suggestion
		label: __( "I'd rather not answer" ),
```

@dlind1 If anyone needs a 20% project idea, making the bot provide these as commitable suggestions would be handy. (Mind the escaping `'I'd'`)

dlind1 replied:
> @dlind1 If anyone needs a 20% project idea, making the bot provide these as commitable suggestions would be handy. (Mind the escaping 'I'd')

Good idea, I've added it to our project planning board 👍

a8ci18n commented on client/signup/steps/woocommerce-install/step-business-info/revenue-options.js:
```diff
@@ -0,0 +1,124 @@
+	if ( ! max ) {
+		return sprintf(
+			/* translators: store product count or revenue */
+			_x( '%s+', 'store product count or revenue', 'woocommerce-admin' ),
```
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

lsl replied:
```suggestion
			/* translators: number followed by a plus, a numerical range estimation e.g. "100+"  */
			_x( '%s+' ),
```

I'm 50-50 on whether we need context here, leaning towards dropping it, I don't think there are any situation/locales the translation would change based on input (number or something else.)

a8ci18n commented on client/signup/steps/woocommerce-install/step-business-info/revenue-options.js:
```diff
@@ -0,0 +1,124 @@
+
+	return sprintf(
+		/* translators: store product count or revenue range */
+		_x( '%1$s - %2$s', 'store product count or revenue range', 'woocommerce-admin' ),
```
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

lsl replied:
```suggestion
		/* translators: numerical range e.g. "100 - 200" */
		_x( '%1$s - %2$s' ),
```
Similar thinking, can probably get by without the context.

lsl commented:
This looks pretty good to me. Ready for review? If you want a bit of a simplification I'd drop the `getNumberRangeString` function and just inline the 3 translation repeats that use it. The `%s+` string doesn't look like it gets hit at runtime. (not sure if you're planning to use this for product count dropdown or not, seems that way?)

allilevine commented:
We need to move the address step before the business info step to have country data for the revenue dropdown. Issue here: https://github.com/Automattic/wp-calypso/issues/60218

Rolilink committed:
woop: fixes currency conversion
Currency conversion was giving up odd results, still don't know what the previous formula tried to accomplish but this one works better.

Rolilink commented on client/signup/steps/woocommerce-install/step-business-info/revenue-options.js:
```diff
@@ -0,0 +1,124 @@
+	const digits = exchangeRate.toString().split( '.' )[ 0 ].length;
+	const multiplier = Math.pow( 10, 2 + digits );
+
+	return Math.round( ( value * exchangeRate ) / multiplier ) * multiplier;
```
Can't figure out what are we trying to do with this formula but I am getting weird conversions, for example `up-to-2500` with Panama as a Country defaults to the US region and should return `up to 2500` but is returning `up to 3000`.

Rolilink replied:
it seems in the conversion it does `(2500 * 1) / 1000 = 2.5` and rounding that up would be `3` then `3 * 1000 = 3000`.

lsl replied:
Math isn't my strong suit but this looks like its doing a smoothing of the numbers so they don't look so random. It's rounding to a number of digits based on the exchange rate.

E.g. MX address,

**Without rounding** 
![Screenshot_2022-01-21_15-38-28](https://user-images.githubusercontent.com/811776/150467530-ad2e8965-592f-455b-9904-e0fa04c86bce.png)

**With rounding**
![Screenshot_2022-01-21_15-35-26](https://user-images.githubusercontent.com/811776/150467538-e12a6709-0f54-4bb1-a183-c034105918fa.png)

The defaulting to US region but still rounding is a bug, it should be returning `value` when that happens instead of applying the smoothing to the USD value.

The bigger issue with this is that we render values in `$` dollars instead of the currency for the value we've converted to (it shouldn't be using the locale currency symbol which we might be using?). We should probably also render USD instead of `$` when we fallback against it too.

lsl replied:
^ should all be fixed now.

allilevine commented:
This needs:
- [ ] Address i18n comments
- [ ] Round options to nearest tenth or hundred.
- [ ] Add EU Countries to the region array (hard-coded list).

lsl committed:
i18n

lsl committed:
Bring back rounding with bugfix

a8ci18n commented on client/signup/steps/woocommerce-install/step-business-info/revenue-options.js:
```diff
@@ -0,0 +1,125 @@
+	},
+	{
+		value: 'rather-not-say',
+		label: __( "I'd rather not say" ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'I'd rather not answer' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-revenue-dropdown)

lsl committed:
Show currency symbols based on store address and currency converted to instead of rendering everything in the users country

lsl committed:
Add euro country codes

lsl committed:
WIP: euro country codes need improving

lsl committed:
Bot reco

lsl commented:
This needs:

- [x] Address i18n comments
- [x] Round options to nearest tenth or hundred.
- [ ] Add EU Countries to the region array (hard-coded list). (WIP - [see code](https://github.com/Automattic/wp-calypso/pull/60066/commits/c1fd46fdad2df5d5d34367f7b02198fc5aa3e4e9#diff-6653f1d7576b7d244b16c279865f265cd1e1a57dde1a2d975116fa901e196892R66))

obenland committed:
Use Woo's list of EU countries

obenland committed:
Use __() when not providing context

Rolilink commented on client/signup/steps/woocommerce-install/step-business-info/index.tsx:
```diff
@@ -40,6 +45,10 @@ export default function StepBusinessInfo( props: WooCommerceInstallProps ): Reac
 		updateOnboardingProfile( 'selling_venues', venue );
 	}
 
+	function updateRevenue( venue: string ) {
```
any reason why the parameter is called venue (?) shouldn't it be revenue? not a big deal tho

Rolilink committed:
fixes variable naming

DustyReagan reviewed (approved):
Works well! 👍

I noticed we're not converting to Yen, for example. Expanding currency support might be a good follow up.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7094264

Hi @obenland, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `What's your current annual revenue?`
- `%s+`
- `%s (I'm just getting started)`
- `Up to %s`
- `More than %s`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7094264).

lsl commented:
> I noticed we're not converting to Yen, for example. Expanding currency support might be a good follow up.

Added https://github.com/Automattic/wp-calypso/issues/60655

We should also be specifying USD if it's falling back to showing USD when not showing it for a US locale. For example it will fallback to USD values for Australia who among many countries, also uses the dollar symbol.

Added https://github.com/Automattic/wp-calypso/issues/60654
